TST: devdeps job found unicode field and said please report

Example log: https://github.com/astropy/specutils/actions/runs/25771983312/job/75696865796

```
Package versions: 
Numpy: 2.5.0.dev0+git20260508.3bea241
Scipy: 1.18.0.dev0+git20260511.1629f86
Matplotlib: 3.12.0.dev58+g2d219875c
astropy: 8.1.0.dev53+g0df5037c7
gwcs: 1.0.4.dev57
asdf: 5.3.1.dev48+g080b06265
asdf-astropy: 0.12.0.dev12+g2bfac8198
stdatamodels: 5.1.0.dev60+g37474e8f1
ndcube: 2.4.0
spectral-cube: 0.6.8.dev15+g87939af93
...
________________________ test_astra_no_model_flux_fail _________________________

    def test_astra_no_model_flux_fail():
        """Test astra loader fail when no model flux is present in the HDUList"""
    
        tmpfile = "astra-temp.fits"
        hdul = mwm_HDUList([0, 0, 1, 0], True, astra=True)
        for hdu in hdul[1:]:
            tt = Table(hdu.data)
            tt.remove_column("model_flux")
            hdu.data = tt.as_array()
>       hdul.writeto(tmpfile, overwrite=True)

specutils/io/default_loaders/tests/test_sdss_v.py:860: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/fits/hdu/hdulist.py:1067: in writeto
    hdu._writeto(hdulist._file)
astropy/io/fits/hdu/base.py:646: in _writeto
    self._writeto_internal(fileobj, inplace, copy)
astropy/io/fits/hdu/base.py:652: in _writeto_internal
    data_offset, data_size = self._writedata(fileobj)
                             ^^^^^^^^^^^^^^^^^^^^^^^^
astropy/io/fits/hdu/base.py:583: in _writedata
    size += self._writedata_internal(fileobj)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.io.fits.hdu.table.BinTableHDU object at 0x7fe259edc7a0>
fileobj = <astropy.io.fits.file._File <_io.BufferedWriter name='astra-temp.fits'>>

    def _writedata_internal(self, fileobj):
        size = 0
    
        data = self.data
        if data is None:
            return size
    
        if _has_unicode_fields(data):
>           raise RuntimeError("Found unicode fields. Please report.")
E           RuntimeError: Found unicode fields. Please report.

astropy/io/fits/hdu/table.py:991: RuntimeError
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -5,7 +5,7 @@ New Features
 ^^^^^^^^^^^^
 
 - Added new SDSS-V ``Spectrum1D`` and ``SpectrumList`` default loaders for
-  ``astraStar`` and ``astraVisit`` model spectra datatypes. [#1325]
+  ``astraStar`` and ``astraVisit`` model spectra datatypes. [#1325, #1330]
 
 Bug Fixes
 ^^^^^^^^^

--- a/specutils/io/default_loaders/tests/test_sdss_v.py
+++ b/specutils/io/default_loaders/tests/test_sdss_v.py
@@ -853,10 +853,10 @@ def test_astra_no_model_flux_fail():
 
     tmpfile = "astra-temp.fits"
     hdul = mwm_HDUList([0, 0, 1, 0], True, astra=True)
-    for hdu in hdul[1:]:
-        tt = Table(hdu.data)
+    for hdu_i in range(1, len(hdul)):
+        tt = Table(hdul[hdu_i].data)
         tt.remove_column("model_flux")
-        hdu.data = tt.as_array()
+        hdul[hdu_i] = fits.BinTableHDU(tt)
     hdul.writeto(tmpfile, overwrite=True)
 
     with pytest.raises(OSError):

